File: Data Structures/Stack&Queues/stacks.py
# Stacks - LIFO (can be implemented like a dynamic array)
stk = []  # Initialize an empty stack (O(1))

# Push elements onto the stack
stk.append(1)  # O(1)
stk.append(2)  # O(1)
stk.append(3)  # O(1)

print(stk)  # Output entire stack (O(n))

# Remove element from stack (pop operation)
x = stk.pop()  # O(1), as it removes the last element
print(x)
print(stk)  # Output entire stack after pop (O(n))

# Display the top element of stack
print(stk[-1])  # O(1), accessing the last element by index

# Search for an element in stack
if 1 in stk:  # O(n), as it searches through the entire stack if necessary
    print(True)

File: Data Structures/Stack&Queues/stacks.cpp
#include <iostream>
#include <stack>
int main(){
std::stack<int> s;

//push an element into stack
s.push(1);
s.push(2);
s.push(3);

//get size of stack
std::cout << "size: " << s.size() << "\n";

//pop from the stack 
s.pop();

//view top element of stack
std::cout << "top: " << s.top() << "\n";
return 0;
}
